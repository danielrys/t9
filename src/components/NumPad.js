// @flow
import * as React from "react"
import { View, StyleSheet } from "react-native"

// components
import PadKey from "./PadKey"

// theme
import Metrics from "../themes/Metrics"

const NUMBERS_CONFIG = [
  {
    value: 1,
    subvalue: "",
  },
  {
    value: 2,
    subvalue: "ABC",
  },
  {
    value: 3,
    subvalue: "DEF",
  },
  {
    value: 4,
    subvalue: "GHI",
  },
  {
    value: 5,
    subvalue: "JKL",
  },
  {
    value: 6,
    subvalue: "MNO",
  },
  {
    value: 7,
    subvalue: "PQRS",
  },
  {
    value: 8,
    subvalue: "TUV",
  },
  {
    value: 9,
    subvalue: "WXYZ",
  },
]

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "space-around",
    alignItems: "center",
    paddingTop: Metrics.spacings.medium,
    maxWidth: Metrics.padKey.baseDimension * 3.5,
    alignSelf: "center",
  },
})

type NumPadProps = {
  onNumberPress: (number | string) => void,
  onDeletePress: any => void,
}

export default class InputBox extends React.PureComponent<NumPadProps> {
  render() {
    const { onNumberPress, onDeletePress } = this.props
    return (
      <View style={styles.container}>
        {NUMBERS_CONFIG.map(keyConfig => (
          <PadKey
            keyConfig={keyConfig}
            onPress={onNumberPress}
            key={keyConfig.value}
          />
        ))}
        <PadKey
          keyConfig={{ value: "<", subvalue: "DEL" }}
          onPress={onDeletePress}
        />
      </View>
    )
  }
}
