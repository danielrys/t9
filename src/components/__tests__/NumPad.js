import React from "react"
import renderer from "react-test-renderer"

// components
import NumPad from "../NumPad"

describe("NumPad component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<NumPad onNumberPress={() => undefined} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
