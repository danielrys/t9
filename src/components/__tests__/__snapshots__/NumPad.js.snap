// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumPad component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-around",
      "maxWidth": 280,
      "paddingTop": 15,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      1
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      2
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      ABC
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      3
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      DEF
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      4
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      GHI
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      5
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      JKL
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      6
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      MNO
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      7
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      PQRS
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      8
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      TUV
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      9
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      WXYZ
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(150,150,150)",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 15,
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0,0,0)",
          "fontSize": 24,
        }
      }
    >
      &lt;
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(50,50,50)",
          "fontSize": 12,
        }
      }
    >
      DEL
    </Text>
  </View>
</View>
`;
